c.c

```diff
@@ -3522,9 +3528,19 @@ wasmCWriteImplementationFile(
     bool parallel = singleFile == NULL;
     FILE* file = singleFile;
 
+    U32 endIndex = startFunctionIndex + functionsPerFile;
+    if (endIndex > functionCount) {
+        endIndex = functionCount;
+    }
+
+    /* Do not create empty files */
+    if (startFunctionIndex > endIndex) {
+        return true;
+    }
+
     if (parallel) {
         char filename[2048];
-        sprintf(filename, "%d.c", fileIndex);
+        sprintf(filename, "%05d.c", fileIndex);
```

Comment:
Is it necessary to limit the filename length here?

Reply:
It's not limiting, it's extending with extra 0s on the left. This will align files in the directory view